### ISSUE TITLE
email provider - can't close smtp connection

### ISSUE THREAD

**Issue by lon4ik4ik** [umputun/remark42#688]
When running the DEV BE solution, I was trying to use the email provider by setting the following values in the compose-dev-backend.yml file:
   - SMTP_HOST
   - SMTP_USERNAME
   - SMTP_PASSWORD
   - SMTP_PORT
   - SMTP_TLS

I could see in the DEBUG logs the following error:

remark42-dev | 2020/04/15 06:40:43.788 [WARN]  {sender/email.go:76 sender.(*Email).Send.func1} can't close smtp connection, close tcp 172.19.0.2:49024->78.46.5.205:587: use of closed network connection
remark42-dev | 2020/04/15 06:40:43.788 [INFO]  {rest/httperrors.go:40 rest.SendErrorJSON} failed to send confirmation - failed to auth to smtp mail.your-server.de:587: unencrypted connection - 500 - 172.19.0.1 - /auth/email/login?id=remark&user=Eli&address=XXXXXX@gmail.com&site=remark [caused by auth/provider/verify.go:188 provider.VerifyHandler.sendConfirmation]
remark42-dev | 2020/04/15 06:40:43.788 [INFO]  {logger/logger.go:122 logger.(*Middleware).Handler.func1.1} GET - /auth/email/login?id=remark&user=Eli&address=XXXXX@gmail.com&site=remark - 1539deba4a54 - 500 (40) - 21.68973ms

Any idea what might be the root cause here?

**Comment by paskal** [comment]
> failed to auth to smtp mail.your-server.de:587: unencrypted connection
My bet is port 587 doesn't support TLS encryption. Please try different ports like `465` if they are available and see if they will work.

**Comment by lon4ik4ik** [comment]
Hi @paskal ,

just tried that while you were writing the comment and in the BE logs seems clean:

remark42-dev | 2020/04/15 08:55:14.085 [DEBUG] {sender/email.go:60 sender.(*Email).Send} send "\n<!DOCTYPE html>\n<html>\n<head>\n\t<meta name=\"viewport\" content=\"width=device-width\" />\n\t<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" />\n</head>\n<body>\n<div style=\"text-align: center; font-family: Arial, sans-serif; font-size: 18px;\">\n\t<h1 style=\"position: relative; color: #4fbbd6; margin-top: 0.2em;\">Remark42</h1>\n\t<p style=\"position: relative; max-width: 20em; margin: 0 auto 1em auto; line-height: 1.4em;\">Confirmation for <b>Eli</b> on site <b>remark</b></p>\n\t<div style=\"background-color: #eee; max-width: 20em; margin: 0 auto; border-radius: 0.4em; padding: 0.5em;\">\n\t\t<p style=\"position: relative; margin: 0 0 0.5em 0;\">TOKEN</p>\n\t\t<p style=\"position: relative; font-size: 0.7em; opacity: 0.8;\"><i>Copy and paste this text into “token” field on comments page</i></p>\n\t\t<p style=\"position: relative; font-family: monospace; background-color: #fff; margin: 0; padding: 0.5em; word-break: break-all; text-align: left; border-radius: 0.2em; -webkit-user-select: all; user-select: all;\">eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZW1hcmsiLCJleHAiOjE1ODY5NjA3MTQsImlzcyI6InJlbWFyazQyIiwibmJmIjoxNTg2OTU4ODU0LCJoYW5kc2hha2UiOnsiaWQiOiJFbGk6OmxvbjRpay5wb2xldm95QGdtYWlsLmNvbSJ9fQ.DeOuPf77pb_FAZ61-a0DgjBK-VFozexDuQ9G7MzQyZY</p>\n\t</div>\n\t<p style=\"position: relative; margin-top: 2em; font-size: 0.8em; opacity: 0.8;\"><i>Sent to lon4ik.polevoy@gmail.com</i></p>\n</div>\n</body>\n</html>\n" to lon4ik.polevoy@gmail.com

But I did not receive the mail and the FE threw an error:
![image](https://user-images.githubusercontent.com/5412350/79346257-70748c00-7f32-11ea-95d8-fbc2fc9d829a.png)

so basically ended with a timeout: 504 Gateway Time-out

I guess not much you can do, I will try to play more with TLS/SSL and see if I around it.

**Comment by paskal** [comment]
You're missing error message in the output above I believe: there should be a log message with an error somewhere message about message being sent.

**Comment by lon4ik4ik** [comment]
Hi @paskal 
I see that after long time:
remark42-dev | 2020/04/15 09:32:19.133 [INFO]  {rest/httperrors.go:40 rest.SendErrorJSON} failed to send confirmation - failed to make smtp client: failed to dial: EOF - 500 - 172.19.0.1 - /auth/email/login?id=remark&user=lon4ik4ik&address=XXXXXXX@gmail.com&site=remark [caused by auth/provider/verify.go:188 provider.VerifyHandler.sendConfirmation]
remark42-dev | 2020/04/15 09:32:19.133 [INFO]  {logger/logger.go:122 logger.(*Middleware).Handler.func1.1} GET - /auth/email/login?id=remark&user=lon4ik4ik&address=XXXXXXXX@gmail.com&site=remark - 1539deba4a54 - 500 (40) - 5m0.017015256s
remark42-dev | 2020/04/15 09:32:19.133 [WARN]  {lgr/adaptor.go:16 lgr.(*Writer).Write} http: superfluous response.WriteHeader call from github.com/go-chi/chi/middleware.Timeout.func1.1.1 (timeout.go:40)

**Comment by lon4ik4ik** [comment]
Ok now is working! thanks for the hints
